reconcile/gitlab_owners.py

```diff
@@ -45,6 +45,18 @@ def __init__(self, gitlab_client, merge_request, owners, dry_run, persistent_lgt
             top_commit = next(commits)
             self.top_commit_created_at = dateparser.parse(top_commit.created_at)
 
+    def expand_groups(self, owners: list[str] | None) -> list[str] | None:
+        if owners:
+            members = set()
+            for name in owners:
+                if group := self.gitlab.get_group_if_exists(name):
+                    members.update(self.gitlab.get_group_members(group))
```

Comment:
even with cache, this will send too many requests to gitlab, should there be a special name pattern for group? Like `group:xxx`

Reply:
trying to be consistent with CODEOWNERS, which accept a group name. example: https://gitlab.cee.redhat.com/releng/konflux-release-data/-/blob/75f40dbc65dfa543a32f51b0d8b8f294734a0e6b/CODEOWNERS#L258

Reply:
do you have other ideas? i understand this is a lot of calls

Reply:
what about https://github.com/app-sre/qontract-reconcile/pull/4906/commits/26321a876eb07aafca1bac2e1d612953df6f02be?

Reply:
also added a condition that there are lgtms in https://github.com/app-sre/qontract-reconcile/pull/4906/commits/3d4244dc292f88e8a7cf5a102d562a50796c75ca